src/component/JoinList.js

```diff
@@ -2,31 +2,32 @@ import React, { useState } from 'react';
 import { withFirestore } from 'react-firestore';
 import { Redirect, Link } from 'react-router-dom';
 
-const JoinList = ({ firestore }) => {
-  const [token, setToken] = useState('');
+const JoinList = ({ token, setToken, firestore }) => {
   const [redirect, setRedirect] = useState(false);
 
-  const renderRedirect = () => {
-    if (redirect) {
-      return <Redirect to="/list" />;
-    }
-  };
-
-  // checks if token exists and if it does, sends to that specific tokens list page, and adds token to local storage
-  //if token does ot exist, alert pops up telling you that
+  // checks if token exists and if it does, sends to that specific tokens
+  // list page, and adds token to local storage if token does ot exist,
+  // alert pops up telling you that
   const checkTokenExists = token => {
-    firestore
-      .collection('lists')
-      .doc(token)
-      .get()
-      .then(doc => {
-        if (doc.exists) {
-          addToLS(token);
-          setRedirect(true);
-        } else {
-          alert('Enter a valid share code and try again.');
-        }
-      });
+    // This conditional allows the Join List button to be accidentally clicked
+    // when the input is blank without throwing an error. Is there a better
+    // fix? - ethan
+    if (token === null) {
+      return;
+    } else {
+      firestore
+        .collection('lists')
+        .doc(token)
+        .get()
+        .then(doc => {
+          if (doc.exists) {
+            addToLS(token);
```

Comment:
Could we change this function name to something more descriptive? What does "LS" stand for?

Reply:
Changed this to addToLocalStorage.

Reply:
Excellent, thank you!